Add lualatex option to compilers

PHPirates opened the pull request:
This PR depends on #732.

#### Additions
Added lualatex as compiler option. And tested that it works on a sample file:

```latex
\documentclass{article}
\usepackage{luacode}

\begin{document}
    A random number:
    \directlua{tex.print(math.random())}
    and another one:
    \begin{luacode}
        tex.print(math.pi)
    \end{luacode}
\end{document}
```

This doesn't automatically select lualatex as compiler, see #737.

RubenSchellekens commented on src/nl/rubensten/texifyidea/run/BibtexSettingsEditor.kt:
```diff
@@ -26,11 +26,13 @@ class BibtexSettingsEditor(private val project: Project) : SettingsEditor<Bibtex
     private lateinit var mainFile: LabeledComponent<TextFieldWithBrowseButton>
-    private lateinit var auxFile: LabeledComponent<TextFieldWithBrowseButton>
+    // Keep track of the the working directory for bibtex, i.e., where bibtex
+    // should find the files it needs. User sets this in latex run configuration.
```
Change to KDoc comment.

RubenSchellekens commented on src/nl/rubensten/texifyidea/run/LatexCompiler.java:
```diff
@@ -48,15 +49,39 @@
+                }
+            }
+
+        } else if (this == LUALATEX) {
```
```suggestion
        }
        else if (this == LUALATEX) {
```

See [code conventions](https://github.com/Ruben-Sten/TeXiFy-IDEA/blob/master/CONTRIBUTING.md#code-conventions).

RubenSchellekens replied:
Oh and remove the empty line above

RubenSchellekens commented on src/nl/rubensten/texifyidea/run/LatexCompiler.java:
```diff
@@ -48,15 +49,39 @@
+            // But lualatex has no -aux-directory
         }
 
         // Custom compiler arguments specified by the user
```
getCommand has become too long. Please refactor into multiple methods :)

RubenSchellekens commented on src/nl/rubensten/texifyidea/run/compiler/BibtexCompiler.kt:
```diff
@@ -23,11 +23,14 @@ internal object BibtexCompiler : Compiler<BibtexRunConfiguration> {
             runConfig.compilerArguments?.let { addAll(it.split("""\s+""".toRegex())) }
 
+            // Include files from auxiliary directory on Windows
+            if (System.getProperty("os.name").contains("Windows")) {
```
Not really that important, but maybe add a `System#isPlatform` extension with a platform enumeration. String checking feels.. hacky?

PHPirates commented:
Thanks for the extensive review :) 

- I added `\directlua` to the autocomplete (apparently it doesn't require the `luacode` package). It seems to be the only important command added by compiling with lualatex.

- I added the `luacode` environment and package, although it seems a bit arbitrary we're including this package but not others.

I hope I did what you intended for the other fixes and the `getPlatformType()` method.

RubenSchellekens commented:
> I added the luacode environment and package, although it seems a bit arbitrary we're including this package but not others.

True. It's better than nothing. Ideally we have indexing of libraries.. See issue #2....
